sycl/include/sycl/ext/oneapi/experimental/device_architecture.hpp

```diff
@@ -24,92 +29,125 @@ enum class architecture {
   //   - the unique ID of the new architecture in SYCL RT source code to support
   //     querying the device architecture
   //
-  x86_64,
-  intel_cpu_spr,
-  intel_cpu_gnr,
-  intel_gpu_bdw,
-  intel_gpu_skl,
-  intel_gpu_kbl,
-  intel_gpu_cfl,
-  intel_gpu_apl,
+  // Important note about keeping architecture IDs below unique:
+  //   - the architecture ID must be a hex number with 16 digits
+  //   - the architecture ID must suit the following template:
+  //     0x AA BBBBBBBBBBBB CC (without spaces), where
+  //       - AA is 2-digit ID of category which must be unique
+  //       - BBBBBBBBBBBB is 12-digit number of architecture itself. It must be
+  //         unique for all architectures inside the category
```

Comment:
```suggestion
  //       - BBBBBBBBBBBB is 12-digit number of architecture itself. It must be
  //         unique for all architectures inside the family
```

If I were you, I'd reserve only 8 digits (32 bits) for the architecture.  This is enough to store the GMDID for Intel GPUs, and it seems sufficient for all the devices we currently know about.  This will leave an addition 4 digits (16 bits) for future use.

I could imagine using those 16 bits for encoding the category of each ID in the future.  Your current strategy assumes that all the IDs in a category are in a distinct range of min / max values.  This seems like a reasonable assumption, and I think it might always be true.  However, it's possible that we may end up with some categories that are not all in a single range of IDs.  In this case, you could use some of these extra bits as an index into a category table.

There might be other uses for the bits in the future too.

Reply:
Thanks for the suggestion! I reserved 16 bits for encoding category to the left of architecture ID, or you meant they should be at the very end?